pallets/collator-assignment/src/lib.rs

```diff
@@ -236,6 +243,15 @@ pub mod pallet {
                     max_collators: collators_per_parathread,
                 });
             }
+
+            // Prioritize paras by tip
+            // As of now this doesn't distinguish between parachains and parathreads
+            // TODO apply different logic to parathreads
+            chains.sort_by(|a, b| {
```

Comment:
I think it would be better if we only sort the chains by tip when we detect congestion as tip is only being paid if we detect congestion.

Reason being that, the first chains on the list gets more collators if there are surplus collators as per `select_chains_with_collators`. So, if there is no congestion in the network, the chains which declared the max tip do not pay fee and based on the max tip setting alone, could be given priority in getting surplus collators.  Even for chains with tip declared, the chain with less max tip will be given lower priority than chain with more tip declared. 

Alternatively, you could always charge the tip and give them more collators even in case there is no congestion. That way there is a tangible cost to be ahead in the list.

Reply:
I think we should not charge tip if there is no congestion. So I agree on sorting only if congestion is detected

Reply:
You're right that it's better to only sort on congestion. Just pushed a new commit with that check